hs_core/templates/search/search.html

```diff
@@ -106,134 +95,398 @@ <h4 class="panel-title">
+                                {% endif %}&laquo; Previous
+                                {% if page_obj.has_previous %}</a>{% endif %}
+                                |
+                                {% if page_obj.has_next %}
+                                    <a href="{{ request.get_full_path|clean_pagination_url }}&amp;page={{ page_obj.next_page_number }}">
+                                {% endif %}Next &raquo;{% if page_obj.has_next %}</a>{% endif %}
+                            </div>
+                        {% endif %}
+                    </div>
+                    <div id="map-view" class="tab-pane fade">
+                        <div id="discover-map" style="width:850px; height:650px"></div>
+                    </div>
+                </div>
             {% endif %}
         </div>
     </div>
 </div>
 {% endblock %}
 
-
 {% block extra_js %}
+
+    <!--<script async defer src="https://maps.googleapis.com/maps/api/js?libraries=drawing"></script>-->
+    <script src="http://maps.googleapis.com/maps/api/js?v=3" type="text/javascript"></script>
+    <script src="http://google-maps-utility-library-v3.googlecode.com/svn/tags/keydragzoom/2.0.8/src/keydragzoom_packed.js" type="text/javascript"></script>
+    <script src="https://googlemaps.github.io/js-marker-clusterer/src/markerclusterer.js" type="text/javascript"></script>
     <script type="text/javascript">
-    var reorderDivs = function() {
-         var faceted_fields = ['author', 'subjects', 'resource_type', 'public', 'owners_names', 'discoverable'];
-         var div_ordering = [];
-         faceted_fields.forEach(function(field){
-             var faceting_div = "faceting-"+field;
-             div_ordering.push(faceting_div);
-         });
-         var i;
-         for (i = 1; i < div_ordering.length; i++) {
-             var div0 = "#" + div_ordering[i-1];
-             var div1 = "#" + div_ordering[i];
-             $(div1).insertAfter(div0);
-         }
-    }
-
-    var clearAllFaceted = function() {
-        localStorage.clear();
-        var textSearch = $("#id_q").val();
-        var clearURL = "/search/?q=" + textSearch;
-        window.location = clearURL;
-    }
-
-    $(document).ready(function () {
-         $("#id_q").attr('placeholder', 'Search...');
-         reorderDivs();
-         $('.collapse').on('shown.bs.collapse', function(){
-            $(this).parent().find(".glyphicon-plus").removeClass("glyphicon-plus").addClass("glyphicon-minus");
-         }).on('hidden.bs.collapse', function(){
-             $(this).parent().find(".glyphicon-minus").removeClass("glyphicon-minus").addClass("glyphicon-plus");
-         });
-
-         $("#id_q").keypress(function(event) {
-            if (event.which == 13) {
-                event.preventDefault();
-                localStorage.clear();
-                $("form").submit();
+        var minFloatingNumber = 0.0001;
+        var map = null;
+        var initMap = function(json_results) {
```

Comment:
All these globals should really be wrapped in a closure to keep from polluting the global namespace.